src/responses.rs

```diff
@@ -15,22 +18,51 @@ pub struct OidcDiscoveryResponse {
 
 #[derive(Deserialize, Debug)]
 /// [JWKs response](https://tools.ietf.org/html/rfc7517)
+/// Contains a list of keys that are retrieved from the jwks uri
 pub struct JWKsResponse {
     /// The keys of the jwks response, see `JWK`
-    pub keys: Vec<JWK>,
+    pub keys: Vec<JsonWebKey>,
 }
 
+/// [JWK](https://tools.ietf.org/html/rfc7517)
+/// Define the structure of each key type that are retrieved from the jwks uri
 #[derive(Deserialize, Debug)]
-/// A single key of the jwks response
-pub struct JWK {
-    /// The key type
-    pub kty: String,
-    /// The key algorithm
-    pub alg: String,
-    /// The Public Keys Component n, the modulus
-    pub n: String,
-    /// The Public Keys Component e, the exponent
-    pub e: String,
+#[serde(untagged)]
```

Comment:
Does [internally tagged](https://serde.rs/enum-representations.html#internally-tagged) with `tag="kty"` not fit better here? Just have to name the enum variants correctly or [rename them](https://serde.rs/variant-attrs.html#variant-attributes) with the `rename` attribute.

Reply:
@miwig I think i would have to use `alg` (RS256) and not `kty` (RSA). When i use `alg`, the compiler will say: "variant field name `alg` conflicts with internal tag".

Reply:
Right, I think you can't have the tag also be a field in the variant, but you can just delete the field since the value is now used to select the right variant.

Concerning `alg` vs `kty`: Unfortunately `alg` is an [optional field in the JWK spec](https://datatracker.ietf.org/doc/html/rfc7517#section-4.4), but since this can be a [security issue](https://portswigger.net/web-security/jwt/algorithm-confusion) (by reusing a public key as a symmetric key), I think it is reasonable to require it. Here [is a list of possible `alg` values.](https://www.iana.org/assignments/jose/jose.xhtml#web-signature-encryption-algorithms) I think `HS256` would be a good candidate to implement to test support for multiple key types.

Reply:
So, you would recommend to use kty over alg? @miwig

Reply:
On the other hand, OIDC discovery with `HS256` doesn't really make sense, as the (symmetric) key must remain secret. :rofl:

Reply:
> So, you would recommend to use kty over alg? @miwig

No. The spec says that `alg` is optional, but this can cause security issues, so I think making `alg` required here should be fine.

Reply:
Ok, are you ok with the changes that i made?

Reply:
LGTM